src/java/org/apache/cassandra/auth/IAuthenticator.java

```diff
@@ -179,5 +195,64 @@ default boolean shouldSendAuthenticateMessage()
         {
             return true;
         }
+
+        /**
+         * @return The assumed mode of authentication attempted using this negotiator, this will usually be some value
+         * of {@link AuthenticationMode#toString()}} unless an implementor provides their own custom authentication
+         * scheme.
+         */
+        default AuthenticationMode getAuthenticationMode()
+        {
+            return AuthenticationMode.UNAUTHENTICATED;
+        }
+    }
+
+    /**
+     * Known modes of authentication supported by Cassandra's provided {@link IAuthenticator} implementations.
+     */
+    abstract class AuthenticationMode
+    {
+        private final String displayName;
+
+        public AuthenticationMode(@Nonnull String displayName)
+        {
+            this.displayName = displayName;
+        }
+
+        /**
+         * User was not authenticated in any particular way.
+         */
+        public static final AuthenticationMode UNAUTHENTICATED = new AuthenticationMode("Unauthenticated") {};
+
+        /**
+         * User authenticated using a password.
+         */
+        public static final AuthenticationMode PASSWORD = new AuthenticationMode("Password") {};
+
+        /**
+         * User authenticated using a trusted identity in their client certificate.
+         */
+        public static final AuthenticationMode MTLS = new AuthenticationMode("Mtls") {};
```

Comment:
Should we spell out the label?
```suggestion
        public static final AuthenticationMode MTLS = new AuthenticationMode("Mutual TLS") {};
```

Reply:
I think this would break JMX as I am not sure how a metric with a space it its name would behave

Reply:
we should probably check that the display name does not contain space

Reply:
good point, I'm not sure what's the behavior for jmx. But if spaces break JMX, we should add some guards around it

Reply:
I'll test whether spaces work w/ JMX, I suspect it does, but isn't desirable as might require clients to escape;  If it doesn't work, I suggest we throw a RuntimeException in the `AuthenticationMode` constructor. 

As far as the name, regardless of whether spaces are allowed, how about `MututalTLS`?

Reply:
It looks like it does work:

```
beans
...
org.apache.cassandra.metrics:name=ConnectedNativeClients,scope=Encrypted,type=Client
org.apache.cassandra.metrics:name=ConnectedNativeClients,scope=Mutual Tls,type=Client
org.apache.cassandra.metrics:name=ConnectedNativeClients,scope=Password,type=Client
org.apache.cassandra.metrics:name=ConnectedNativeClients,scope=Unencrypted,type=Client
org.apache.cassandra.metrics:name=ConnectedNativeClients,type=Client
```

But it isn't easy to use, and I imagine will give some tooling trouble.   For it to work with jmxterm for example I had to double `\` escape the space:

```
$>bean org.apache.cassandra.metrics:name=ConnectedNativeClients,scope=Mutual Tls,type=Client
#IllegalArgumentException: Please specify domain using either -d option or domain command
$>bean org.apache.cassandra.metrics:name=ConnectedNativeClients,scope=Mutual\\ Tls,type=Client
#bean is set to org.apache.cassandra.metrics:name=ConnectedNativeClients,scope=Mutual Tls,type=Client
$>get *
#mbean = org.apache.cassandra.metrics:name=ConnectedNativeClients,scope=Mutual Tls,type=Client:
Value = 5;
```

I think we should document that spaces in the mode name aren't preferred as it may not work well with some tooling. I will do that!

Reply:
Addressed in [8f082a1](https://github.com/apache/cassandra/pull/3085/commits/8f082a189a5ff5f781265b10d7777d47c1af6cc7) and renamed `Mtls` -> `MutualTls`.  I also realized was still using `String` in `ClientMetrics` instead of `AuthenticationMode` so I updated that as well, hopefully that looks good!